libraries/ms-common/inc/ms_test_helpers.h

```diff
@@ -87,3 +87,8 @@
                        .data = (ack_status) };                                                     \
     TEST_ASSERT_OK(can_ack_handle_msg(&(can_storage).ack_requests, &msg));                         \
   })
+
+// Returns STATUS_CODE_OK if the *value_to_test == value_to_compare within a certain time +- the
+// tolerance
+StatusCode time_assert(uint8_t value_to_compare, uint8_t *value_to_test, double time_ms,
+                       double tolerance_ms);
```

Comment:
I think this function should be prefixed with the module name (`ms_test_helpers_time_assert`) since just seeing `time_assert` in a test you wouldn't know where it came from. Also have you thought about having a `TEST_ASSERT_EQUAL_AT_TIME` macro (or some similar name) that just does `TEST_ASSERT_OK(time_assert(...))`, since that seems to be the common pattern?

Also, since `*value_to_test` changes due to another thread, it should be marked `volatile` (`volatile uint8_t *value_to_test`) so that the compiler doesn't optimize it out.